[NETBEANS-1074] module review websvc.metro.lib

rtaneja1 opened the pull request:
Added missing license.

rtaneja1 commented:
Thanks Junichi, no failures in metro-2.0-license.txt in nbbuild/build/verifylibsandlicenses.xml now.

junichi11 commented:
@rtaneja1 Thanks for fixing it and sorry, Reema. I might be wrong. I realized that the license is CDDL-1.0 in the existing license file but it is CDDL-1.1 in the website. 

@matthiasblaesing Is it OK with CDDL-1.1?

rtaneja1 commented:
Yet to push the latest changes in binaries-list file which now points to jars on maven repos, as have queries regarding licenses for these jars.
<binaries-list>
FDECFB78184C7D19E7E20130A7D7E88C1DF0BDD1 metro-1.4-doc.zip
791D6754F998ECEAABFDD430C72F7AEBE8E3E77D javax.xml:webservices-api:2.1-b16
86D91423DFEE9C5FBFEBB91133035338F04F545F com.sun.xml.ws:webservices-extra:2.1-b16
E2A25D3DC5E77ADA4267E2A00030E01046E94C8F javax.xml:webservices-extra-api:2.1-b16
C22E0867A5297FB8B8D28316EF42F9408F675F79 com.sun.xml.ws:webservices-rt:2.1-b16
DB20F88EE3A516190C3F89929E0618452B087C30 com.sun.tools.ws:webservices-tools:2.1-b16

1. metro-1.4-doc.zip could not find matching zip on maven repo, leave it as it is?

2. webservices-extra.jar, webservices-extra-api.jar only have version 2.1-b16 available. Hence upgraded other jars to same version. Is it fine?

3. webservices-api.jar is CDDL1.0 GPL2.0, which license text should be used for this, nbbuild/licenses/CDDL-1.0?

4. webservices-extra.jar, webservices-extra-api.jar have CDDL GPL license, which license text should be used, nbbuild/licenses has only CDDL-GPL-2CP.

5. webservices-rt.jar and webservices-tools.jar have CDDL-1.0, GPL, GPL 2.0, which license should be used.

Thanks,

matthiasblaesing commented:
Reema thank you for your investigation! In general GPL and Apache Foundation are a bad match, so if there is another license available, that is most probably the correct one. GPLv2 with class path exception might be workable, but apache legal is reluctant. So let's stay away. If a project is dual-licenses under a variant of GPL and CDDL, then choose CDDL, as that license is at least accepted inside the ASF.

For the metro library - I think using one released version is the right approach. I also verified, that for 2.1b16 no javadocs are present on maven central (at least my search-fu was not good enough to find them). But then we have the documentation for 1.4 and the binaries for 2.1.... (not directed at you, just thinking out loud). So you binaries-list looks sensible.

The alternative: Go directly to a recent version of metro 2.3.1 - I think I saw the javadocs for "-rt" on maven central. I think that might be a viable way.

rtaneja1 commented:
Thanks Matthias, corrected binaries-list to point to metro 2.3.1. License used is CDDL1.1 - https://javaee.github.io/glassfish/LICENSE
Also modified libs.amazon/binaries-list as ant verify-libs-and-licenses was failing in downloading two artifacts.

matthiasblaesing commented on websvc.metro.lib/nbproject/project.properties:
```diff
@@ -22,14 +22,18 @@ jnlp.indirect.jars=\
+release.external/webservices-extra-2.3.1-javadoc.jar=docs/webservices-extra-2.3.1-javadoc.jar
+release.external/webservices-tools-2.3.1-javadoc.jar=docs/webservices-tools-2.3.1-javadoc.jar
+release.external/webservices-extra-api-2.3.1-javadoc.jar=docs/webservices-extra-api-2.3.1-javadoc.jar
+release.external/webservices-rt-2.3.1-javadoc.jar=docs/webservices-rt-2.3.1-javadoc.jar
```
Please also update: `websvc.metro.lib/src/org/netbeans/modules/websvc/metro/lib/metro.xml`. That file declares the library in netbeans (Tools -> Library). The javadoc files need to be added to the javadoc volumne (see line 39 for the reference to the metro doc). As the binaries are not versioned, I would also drop the version from the java docs.

When you update `metro.xml` I would also udate the maven dependencies to:

```
org.glassfish.metro:webservices-api:2.3.1
org.glassfish.metro:webservices-extra:2.3.1
org.glassfish.metro:webservices-extra-api:2.3.1
org.glassfish.metro:webservices-rt:2.3.1
org.glassfish.metro:webservices-tools:2.3.1
```

rtaneja1 replied:
Corrected, Thanks

matthiasblaesing commented on libs.amazon/external/binaries-list:
```diff
@@ -15,7 +15,7 @@
+5043BFEBC3DB072ED80FBD362E7CAF00E885D8AE commons-logging:commons-logging:1.1.1
 FD32786786E2ADB664D5ECC965DA47629DCA14BA commons-codec:commons-codec:1.3
-3D74BFB229C259E2398F2B383D5425CB81C643F0 org.apache.httpcomponents:httpclient:4.1.1
+3D1D918F32709E33BA7DDB2C4E8D1C543EBE713E org.apache.httpcomponents:httpclient:4.1.1
```
Good catch!

matthiasblaesing commented:
Thank you - that should work. I'd like you to do final things:

- Adjust your author information in git -- your commits are authored with a valid email address, but the author name is `rtaneja`. That looks like an account name - could you please replace it with your real name?
- Squash the commits before commit

If you agree I feel free to execute and merge to master.

junichi11 commented:
> Squash the commits before commit

I also think we should do it.

Should we increase the specification version?

matthiasblaesing commented:
The update holds the specification version update, however, there are still more commit here, then necessary. The newest commits were squashed, but the old ones not. I suspect, the merge from master makes it difficult to completely squash. This should be resolvable by rebasing the changeset onto current master and squash then.

Please also recheck the author information, it is still looks like an account name.

mkleint commented on websvc.metro.lib/src/org/netbeans/modules/websvc/metro/lib/metro.xml:
```diff
@@ -36,13 +36,23 @@
+                org.glassfish.metro:webservices-extra-api:2.3.1:jar 
+                org.glassfish.metro:webservices-rt:2.3.1:jar 
+                org.glassfish.metro:webservices-tools:2.3.1:jar
+            </value>
```
lgtm, however i don't recall anymore why you should check with me on this one ;)

I suppose when added to a maven project, all dependencies are properly added to pom.xml  and the project builds?

maybe as part of the PR, remove the reference to me.

rtaneja1 replied:
Thanks, done.

rtaneja1 commented:
Thanks Matthias, Corrected commit. Probably it was old commit, git log shows author name correctly
>>commit 130df83d7b59ea63625a6ea0f544538eed15d0ac
>>Author: Reema Taneja <reema.taneja@oracle.com>
>>Date:   Mon Sep 3 22:35:08 2018 -0700

matthiasblaesing commented on websvc.metro.lib/src/org/netbeans/modules/websvc/metro/lib/metro.xml:
```diff
@@ -36,13 +36,22 @@
-        <resource>jar:nbinst://org.netbeans.modules.websvc.metro.lib/docs/metro-1.4-doc.zip!/</resource>
+        <resource>jar:nbinst://org.netbeans.modules.websvc.metro.lib/docs/webservices-api-javadoc.jar!/</resource>
+        <resource>jar:nbinst://org.netbeans.modules.websvc.metro.lib/docs/webservices-extra-javadoc.jar!/</resource>
+        <resource>jar:nbinst://org.netbeans.modules.websvc.metro.lib/docs/webservices-extra-api-javadoc.jar!/</resource-->
```
The closing tag is wrong - \</resource--> should be \</resource>

rtaneja1 replied:
Corrected, Thanks

geertjanw commented:
Great, merging it, thanks for all the work here.
